### ISSUE TITLE
Regression: csv-table with file option is broken if used in included file in sphinx 1.8.1.

### ISSUE THREAD

**Issue by ociule** [sphinx-doc/sphinx#5495]
Subject: csv-table with file option is now broken if used in an included file

### Problem
- index.rst includes a test_table_csv_file.rst
- test_table_csv_file.rst has a csv-file directive with a file option
On sphinx 1.7.x, this works fine
On sphinx 1.8.1, this generates the following error:

```
test_csv_table_file.rst <included from C:\Users\ociule\work\sphinx_test\index.rst>:4: WARNING: Problems with "csv-table" directive path:

[Errno 22] Invalid argument: 'test_csv_table_file.rst <included from C:/Users/ociule/work/sphinx_test/test.csv'.

.. csv-table::
  :file: test.csv
```
#### Procedure to reproduce the problem
Start a new sphinx project.

Add this is index.rst:

```
.. include:: test_csv_table_file.rst
```

Create test_csv_table_file.rst:

```

.. csv-table::
  :file: test.csv

```

Create test.csv with whatever valid csv content, for example:
```
a,b
1,2
```

#### Error logs / results
```
test_csv_table_file.rst <included from C:\Users\ociule\work\sphinx_test\index.rst>:4: WARNING: Problems with "csv-table" directive path:

[Errno 22] Invalid argument: 'test_csv_table_file.rst <included from C:/Users/ociule/work/sphinx_test/test.csv'.

.. csv-table::
  :file: test.csv
```

#### Expected results
In sphinx 1.7.x, this works fine, and includes the csv data as expected into the generated table.

In docutils, testing with `rst2html.py index.rst > index.html`, this also works fine, proving it's not a docutils bug.

I suspect whatever recent changes were made to the code that parses the file path before passing it to docutils.

### Reproducible project / your project

[sphinx_test.zip](https://github.com/sphinx-doc/sphinx/files/2423247/sphinx_test.zip)

### Environment info
- OS: Win 7
- Python version: 3.7
- Sphinx version: 1.8.1

**Comment by tk0miya** [comment]
Note: This bug comes from #4821

**Comment by tk0miya** [comment]
Fixed by #5530.
Thank you for reporting.

**Comment by iugo** [comment]
sphinx 1.8.0, same problem
